### PR TITLE
Adding passthrough behavior for API Gateway integration

### PR DESCRIPTION
**Pull request by iceycake**
Should address https://github.com/hashicorp/terraform/issues/7475 and https://github.com/hashicorp/terraform/issues/7141

### CODE REVIEW

**Comment by stack72** [comment]
Thanks so much for the PR here @iceycake 

Just running the tests now :) Will merge if all goes green!

Paul

File: builtin/providers/aws/resource_aws_api_gateway_integration.go [hunk 1]
```diff
@@ -119,11 +132,12 @@ func resourceAwsApiGatewayIntegrationCreate(d *schema.ResourceData, meta interfa
...
-		RequestTemplates:   aws.StringMap(templates),
-		Credentials:        credentials,
-		CacheNamespace:     nil,
-		CacheKeyParameters: nil,
+		RequestParameters:   aws.StringMap(parameters),
+		RequestTemplates:    aws.StringMap(templates),
+		Credentials:         credentials,
+		CacheNamespace:      nil,
+		CacheKeyParameters:  nil,
+		PassthroughBehavior: aws.String(d.Get("passthrough_behavior").(string)),
```
Comment (stack72) [inline builtin/providers/aws/resource_aws_api_gateway_integration.go]: As this is an optional value and has no default value, we should check to see if it is set before adding to the params

**Comment by stack72** [comment]
Hi @iceycake 

When trying to run the tests, I get the following:

```
% make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSAPIGatewayIntegration_'                   2 ↵
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSAPIGatewayIntegration_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayIntegration_basic
--- FAIL: TestAccAWSAPIGatewayIntegration_basic (40.04s)
	testing.go:264: Step 1 error: Error applying: 1 error(s) occurred:

		* aws_api_gateway_integration.test: Error creating API Gateway Integration: BadRequestException: Invalid passthrough behavior specified. Valid options are WHEN_NO_MATCH, WHEN_NO_TEMPLATES, and NEVER
			status code: 400, request id: de9577d4-5299-11e6-b917-8f8a0247d01c
FAIL
exit status 1
FAIL	github.com/hashicorp/terraform/builtin/providers/aws	40.062s
make: *** [testacc] Error 1
```

Thoughts on this? I also left 1 small comment inline

Paul

**Comment by iceycake** [comment]
@stack72 Sorry about not being test carefully.  Hopefully, this fixed the issue.

**Comment by radeksimko** [comment]
Since the field is not `ForceNew` it will also need to be reflected in the Update() function, otherwise updating passthrough behaviour won't work and people are going to be forced to recreate the whole resource just to update passthrough behaviour. 😉

**Comment by stack72** [comment]
Hi @iceycake 

The changes now allow the tests to pass :)

```
% make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSAPIGatewayIntegration_'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSAPIGatewayIntegration_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayIntegration_basic
--- PASS: TestAccAWSAPIGatewayIntegration_basic (50.23s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/aws	50.242s
```

We just need to make sure that @radeksimko's comments are taken onboard, what happens when an Update to the PassthroughBehaviour is made?

P.

**Comment by radeksimko** [comment]
Ignore my previous comment - I overlooked this piece of code:

```go
 func resourceAwsApiGatewayIntegrationUpdate(d *schema.ResourceData, meta interface{}) error {
 	return resourceAwsApiGatewayIntegrationCreate(d, meta)
 }
```

This LGTM then.

**Comment by stack72** [comment]
ah perfect - thanks @radeksimko :)

@iceycake this LGTM! Merging now :) thanks for all the work here!

**Pull request merged**